zboxcore/sdk/sdk.go

```diff
@@ -636,18 +636,29 @@ type UpdateValidator struct {
 	IsShutdown               *bool             `json:"is_shutdown,omitempty"`
 }
 
-func (v *Validator) ConvertToValidationNode() *blockchain.ValidationNode {
-	return &blockchain.ValidationNode{
+func (v *UpdateValidator) ConvertToValidationNode() *blockchain.UpdateValidationNode {
+	blockValidator := &blockchain.UpdateValidationNode{
 		ID:      string(v.ID),
 		BaseURL: v.BaseURL,
-		StakePoolSettings: blockchain.StakePoolSettings{
-			DelegateWallet: v.DelegateWallet,
-			MinStake:       v.MinStake,
-			MaxStake:       v.MaxStake,
-			NumDelegates:   v.NumDelegates,
-			ServiceCharge:  v.ServiceCharge,
-		},
 	}
+
+	sp := &blockchain.UpdateStakePoolSettings{
+		DelegateWallet: v.DelegateWallet,
+		MinStake:       v.MinStake,
+		MaxStake:       v.MaxStake,
+		NumDelegates:   v.NumDelegates,
+		ServiceCharge:  v.ServiceCharge,
+	}
+
+	if v.DelegateWallet != nil ||
+		v.MinStake != nil ||
+		v.MaxStake != nil ||
+		v.NumDelegates != nil ||
+		v.ServiceCharge != nil {
+		blockValidator.StakePoolSettings = sp
```

Comment:
so stake pool settings will be updated only when all fields are set?

Reply:
No, it will be set when atleast one of the field is set (which means atleast one of the field is changing), otherwise it will be nil.